Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h

```diff
@@ -0,0 +1,253 @@
+#include <CGAL/Bbox_3.h>
+#include <CGAL/Iso_cuboid_3.h>
+#include <CGAL/convex_hull_3.h>
+#include <CGAL/Polyhedron_3.h>
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#ifdef OBB_BENCHMARKS
+#include <CGAL/Timer.h>
+#endif
+
+#if defined(CGAL_EIGEN3_ENABLED)
+#include <CGAL/Eigen_linear_algebra_traits.h>
+#endif
+
+namespace CGAL {
+namespace Optimal_bounding_box {
+
+
+#if defined(CGAL_EIGEN3_ENABLED)
+typedef CGAL::Eigen_linear_algebra_traits Linear_algebra_traits;
+#endif
+
+
+// works on matrices only
+/// \cond SKIP_IN_MANUAL
+template <typename Vertex, typename Matrix>
+void post_processing(const Matrix& points, Vertex& R, Matrix& obb)
+{
+  CGAL_assertion(points.cols() == 3);
+  CGAL_assertion(R.rows() == 3);
+  CGAL_assertion(R.cols() == 3);
+  CGAL_assertion(obb.rows() == 8);
+  CGAL_assertion(obb.cols() == 3);
+
+  // 1) rotate points with R
+  Matrix rotated_points(points.rows(), points.cols());
+  rotated_points = points * Linear_algebra_traits::transpose(R);
+
+  // 2) get AABB from rotated points
+  typedef CGAL::Simple_cartesian<double> K;
+  typedef K::Point_3 Point;
+  std::vector<Point> v_points; // Simplex -> std::vector
+  for(std::size_t i = 0; i < rotated_points.rows(); ++i)
+  {
+    Point p(rotated_points(i, 0), rotated_points(i, 1), rotated_points(i, 2));
+    v_points.push_back(p);
+  }
+  CGAL::Bbox_3 bbox;
+  bbox = bbox_3(v_points.begin(), v_points.end());
+  K::Iso_cuboid_3 ic(bbox);
+
+  // Matrix is [dynamic, 3] at compile time, so it needs allocation of rows at run time.
+  Matrix aabb(8, 3);
+
+  for(std::size_t i = 0; i < 8; ++i)
+  {
+    aabb.set_coef(i, 0, ic[i].x());
+    aabb.set_coef(i, 1, ic[i].y());
+    aabb.set_coef(i, 2, ic[i].z());
+  }
+
+  // 3) apply inverse rotation to rotated AABB
+  obb = aabb * R;
+}
+/// \endcond
+
+/// \ingroup OBB_grp
+/// calculates the optimal bounding box.
+///
+/// @tparam Point the point type
+/// @tparam LinearAlgebraTraits a model of `LinearAlgebraTraits`
+///
+/// @param points the input points that are included in the optimal bounding box.
+/// @param obb_points the eight points of the optimal bounding box to be calculated.
+/// @param use_ch a bool flag to indicating whether to use the convex hull of the input points
+/// as an optimization step.
+template <typename Point, typename LinearAlgebraTraits>
+void find_obb(std::vector<Point>& points,
+              std::vector<Point>& obb_points,
+              LinearAlgebraTraits&,
+              bool use_ch)
+{
+  CGAL_assertion(points.size() >= 3);
+
+  if(obb_points.size() != 8) // temp sanity until the API is decided
+    obb_points.resize(8);
+  CGAL_assertion(obb_points.size() == 8);
+
+  // eigen linear algebra traits
+  typedef typename LinearAlgebraTraits::MatrixXd MatrixXd;
+  typedef typename LinearAlgebraTraits::Matrix3d Matrix3d;
+
+  MatrixXd points_mat;
+
+  // get the ch3
+  if(use_ch)
+  {
+    // find the ch - todo: template kernel
```

Comment:
Let's remove the dependency on the convex hull from obb.h and the API. 

Add a function to the package Convex_hull_3 that given a range of points writes the convex hull points into an output iterator.